src/createAsyncThunk.ts

```diff
@@ -263,6 +267,7 @@ export type AsyncThunkFulfilledActionCreator<
     arg: ThunkArg
     requestId: string
     requestStatus: 'fulfilled'
+    additionalMeta: { [key: string]: unknown }
```

Comment:
Also, these would need to be correctly typed - which might make the whole thing reaaally complicated though :(

Reply:
Agree, it might get complicated, but I think it's an essential feature, because right now I don't see any good ways to identify some particular action in a middleware.

For example, right now we are switching to redux toolkit and we have an ability to identify whether the action should show loader when loading or not. It's a global loader which is handled in a redux middleware, and right now I don't have any ideas of how to implement it with redux toolkit thunks, but changing meta, which is unfortunately impossible.


I could try to type it correctly

Reply:
Just to be sure here: you've seen the matchers and those don't do it for you?
https://redux-toolkit.js.org/api/matching-utilities

Reply:
Yes, I have seen them and they still don't help.

In current code base, we have actions that might have a meta data field 'showLoader'. The middleware fires special actions if there's 'showLoader: true'.

So, I doubt these could help me, because I fire the actions only if there's 'showLoader:true'

Reply:
> I could try to type it correctly

You can try, but please try to keep if as simple (while correct) as possible as these types here already tend to kinda explode. I'd suggest adding the return type for this this to `AsyncThunkConfig` which is probably the least intrusive way to implement this.